mycity/mycity/intents/trash_intent.py

```diff
@@ -76,32 +75,29 @@ def get_trash_day_info(mycity_request):
 
     # currently assumes that trash day is the same for all units at
     # the same street address
-    address = str(a['house']) + " " + str(a['street_full'])
-    zip_code = str(a["other"]).zfill(5) if a["other"] and a["other"].isdigit() else None
-    neighborhood = a["other"] if a["other"] and not a["other"].isdigit() else None
-
-    zip_code_key = intent_constants.ZIP_CODE_KEY
-    if zip_code is None and zip_code_key in \
-            mycity_request.session_attributes:
-        zip_code = mycity_request.session_attributes[zip_code_key]
+    address = " ".join([
+        parsed_address['AddressNumber'],
+        parsed_address['StreetName'],
+        parsed_address['StreetNamePostType']])
+    neighborhood = parsed_address["PlaceName"] \
+        if "PlaceName" in parsed_address \
+        and not parsed_address["PlaceName"].isdigit() \
+        else None
 
     if "Neighborhood" in mycity_request.intent_variables and \
         "value" in mycity_request.intent_variables["Neighborhood"]:
-        neighborhood = mycity_request.intent_variables["Neighborhood"]["value"]
-
+            neighborhood = \
+                mycity_request.intent_variables["Neighborhood"]["value"]
 
     try:
-        trash_days = get_trash_and_recycling_days(address, zip_code, neighborhood)
+        trash_days = get_trash_and_recycling_days(address, neighborhood)
         trash_days_speech = build_speech_from_list_of_days(trash_days)
 
         mycity_response.output_speech = speech_constants.PICK_UP_DAY.format(trash_days_speech)
         mycity_response.should_end_session = True
 
     except InvalidAddressError:
         address_string = address
-        if zip_code:
-            address_string = address_string + " with zip code {}"\
-                .format(zip_code)
         mycity_response.output_speech = speech_constants.ADDRESS_NOT_FOUND.format(address_string)
```

Comment:
If the given address is not in Boston, Alexa will return the `ADDRESS_NOT_FOUND` string and will exit the app.
That constant says: **I can't seem to find {address}. Try another address.**

This can lead to a poor user experience since the user might expect to continue the session and give another address.

Reply:
Good idea! I've created #287 for adding this.